spec/models/coronavirus/sub_section/structured_content_spec.rb

```diff
@@ -0,0 +1,100 @@
+require "rails_helper"
+
+RSpec.describe Coronavirus::SubSection::StructuredContent do
+  let(:content) { "###title \n [test](/coronavirus)" }
+
+  describe "#parseable?" do
+    it "returns true if content is valid" do
+      expect(described_class.parseable?(content)).to be true
+    end
+
+    it "returns false if hashes are missing from title" do
+      content = "title \n [test](/coronavirus)"
+      expect(described_class.parseable?(content)).to be false
+    end
+
+    it "returns false if a link is malformed" do
+      content = "###title \n [test](/coronavirus"
+      expect(described_class.parseable?(content)).to be false
+    end
+
+    it "returns false if there's a plaintext line in content" do
+      content = "###title \n [test](/coronavirus) \n plan text \n [another](/link)"
+      expect(described_class.parseable?(content)).to be false
+    end
+  end
+
+  describe "#parse" do
+    let(:parsed_content) { described_class.parse(content) }
+    it "creates the expected output" do
+      expected_output = [
+        Coronavirus::SubSection::StructuredContent::Header.new("title"),
+        Coronavirus::SubSection::StructuredContent::Link.new("test", "/coronavirus"),
+      ]
+      expect(parsed_content.items).to eq(expected_output)
+    end
+  end
+
+  describe "#error_lines" do
+    it "returns an empty array for valid content" do
+      expect(described_class.error_lines(content)).to eq([])
+    end
+
+    it "returns an error when the header is wrong" do
+      content = "title \n [test](/coronavirus)"
+      expect(described_class.error_lines(content)).to eq(%w[title])
+    end
+
+    it "returns an error when a link is malformed" do
+      content = "###title \n [test](/coronavirus"
+      expect(described_class.error_lines(content)).to eq(%w[[test](/coronavirus])
+    end
+
+    it "can return multiple errors" do
+      content = "title \n [test](/coronavirus"
+      expect(described_class.error_lines(content).length).to eq(2)
+    end
+  end
+
+  describe "#create_content_groups_from_links_and_headings" do
+    it "creates multiple groups given multiple titles" do
+      content = "#title \n [test](/coronavirus) \n #title2 \n [test2](/government)"
+      expected_output = [
+        {
+          title: "title",
+          list: [Coronavirus::SubSection::StructuredContent::Link.new("test", "/coronavirus")],
```

Comment:
I've suggested above that this method should be moved to the presenter. I think that we do that, then the equivalent to this test, can actually test the structure of the list i.e. that it contains `url`s and `label`s

Reply:
done!